### ISSUE TITLE
java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider

### ISSUE THREAD

**Issue by b055man** [flutter/flutter#66108]
[✓] Flutter: is fully installed. (Channel stable, 1.20.3, on Mac OS X 10.15.4 19E287, locale en-GB)
[✓] Android toolchain - develop for Android devices: is fully installed. (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS: is fully installed. (Xcode 11.4.1)
[✓] Android Studio: is fully installed. (version 3.6)
[✓] VS Code: is fully installed. (version 1.48.2)
[✓] Connected device: is fully installed. (2 available)

## Steps to Reproduce

Looks to be related to the one logged here: https://github.com/flutter/flutter/issues/65388 but it has been closed already.

1. Run `flutter create bug`.
2. Update the files as follows: 
- main.dart
```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  TextEditingController _controller;

  @override
  void initState() {
    _controller = TextEditingController();
    _controller.value = TextEditingValue(text: 'abc');
    super.initState();
  }

  @override
  void dispose() {
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            TextField(
              controller: _controller,
            ),
          ],
        ),
      ),
    );
  }
}

```
</details>
3. Build and run the app.

**Expected results:**
App starts without exceptions

**Actual results:**
App logs an exception

<details>
  <summary>Logs: Immediately after starting the app, the following exception is logged: </summary>

```
W/ClipData(29672): Failure opening stream
W/ClipData(29672): java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/file/125 from pid=29672, uid=10307 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()
W/ClipData(29672): 	at android.os.Parcel.createException(Parcel.java:1950)
W/ClipData(29672): 	at android.os.Parcel.readException(Parcel.java:1918)
W/ClipData(29672): 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
W/ClipData(29672): 	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)
W/ClipData(29672): 	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:698)
W/ClipData(29672): 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1460)
W/ClipData(29672): 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1404)
W/ClipData(29672): 	at android.content.ClipData$Item.coerceToText(ClipData.java:351)
W/ClipData(29672): 	at io.flutter.plugin.platform.PlatformPlugin.getClipboardData(PlatformPlugin.java:278)
W/ClipData(29672): 	at io.flutter.plugin.platform.PlatformPlugin.access$700(PlatformPlugin.java:24)
W/ClipData(29672): 	at io.flutter.plugin.platform.PlatformPlugin$1.getClipboardData(PlatformPlugin.java:81)
W/ClipData(29672): 	at io.flutter.embedding.engine.systemchannels.PlatformChannel$1.onMethodCall(PlatformChannel.java:141)
W/ClipData(29672): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:230)
W/ClipData(29672): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
W/ClipData(29672): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
W/ClipData(29672): 	at android.os.MessageQueue.nativePollOnce(Native Method)
W/ClipData(29672): 	at android.os.MessageQueue.next(MessageQueue.java:326)
W/ClipData(29672): 	at android.os.Looper.loop(Looper.java:170)
W/ClipData(29672): 	at android.app.ActivityThread.main(ActivityThread.java:6991)
W/ClipData(29672): 	at java.lang.reflect.Method.invoke(Native Method)
W/ClipData(29672): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
W/ClipData(29672): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:884)
```

**Comment by pedromassangocode** [comment]
Hi @b055man 
Please provide the complete output of your `flutter doctor -v` without modifications.

Cc @Hixie @justinmc this seems related to https://github.com/flutter/flutter/issues/65388 even though the initial log message is different, it also ends with a clipData exception.

**Comment by b055man** [comment]
Adding requested info (in the meantime I upgraded to 1.20.4, but the issue still reproduces).
**One important note**: the app does not crash, it's the log being printed - in case of my app it poluttes the logcat heavily. Apart from that, the field itself seems to work as expected.

flutter doctor -v
[✓] Flutter (Channel stable, 1.20.4, on Mac OS X 10.15.4 19E287, locale en-GB)
    • Flutter version 1.20.4 at /Users/aam074/software/flutter
    • Framework revision fba99f6cf9 (4 days ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/aam074/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.0

[✓] Connected device (1 available)
    • LG H870 (mobile) • xxxx • android-arm64 • Android 9 (API 28)

• No issues found!

**Comment by hamdikahloun** [comment]
Hi @b055man 

- This Exception , happen only with your LG G6 or with all the devices even it's the emulator ?

- If you copy any text from anywhere without using the Clip Tray Feature and open the app, the exception still there?

**Comment by b055man** [comment]
@hamdikahloun 
It looks like you're on the right track 👍 

> This Exception , happen only with your LG G6 or with all the devices even it's the emulator ?

Tried now on a different phone (with Android 10) and emulator - and indeed, it does not happen.

> If you copy any text from anywhere without using the Clip Tray Feature and open the app, the exception still there?

I'm not sure how to copy something without using the Clip Tray (it's my dev phone, not a daily one), but even with it enabled, when I copied some **text**, the exception does not occur. And then I noticed that the problem must have been related to the fact that the last (and only?) item in the Clip Tray was an image...

Tried your workaround and indeed it adresses the issue - a minor sideffect is that when the Clip Tray is updated, it shows a toast 'Item added to the Clip Tray' (or something similar) and the 'Value' itself (pun intentended) is stored in the Clip Tray as first item.

Your comment says this is already addressed in the master channel - are there plans for cherry-picking that into the master or will it have to wait for the next stable release? Can you add a reference to the relevant fix in the master channel maybe?

**Comment by hamdikahloun** [comment]
@b055man I will add new patch to master channel in the next few hours. It will be available soon.
Thanks

**Comment by hamdikahloun** [comment]
@b055man Workaround for stable channel

```
import android.app.ActivityManager;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.os.Build;
import android.util.Log;
import android.view.HapticFeedbackConstants;
import android.view.SoundEffectConstants;
import android.view.View;
import android.view.Window;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java.util.List;
import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.systemchannels.PlatformChannel;
import android.content.ClipDescription;
import static io.flutter.plugin.platform.PlatformPlugin.DEFAULT_SYSTEM_UI;
import static android.content.ClipDescription.MIMETYPE_TEXT_PLAIN;

public class MainActivity extends FlutterActivity
    implements PlatformChannel.PlatformMessageHandler {
  private PlatformChannel.SystemChromeStyle currentTheme;
  private int mEnabledOverlays;

  @Override
  public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
    flutterEngine.getPlatformChannel().setPlatformMessageHandler(this);
  }

  @Override
  public void playSystemSound(@NonNull PlatformChannel.SoundType soundType) {
    if (soundType == PlatformChannel.SoundType.CLICK) {
      View view = getWindow().getDecorView();
      view.playSoundEffect(SoundEffectConstants.CLICK);
    }
  }

  @Override
  public void vibrateHapticFeedback(@NonNull PlatformChannel.HapticFeedbackType feedbackType) {
    View view = getWindow().getDecorView();
    switch (feedbackType) {
      case STANDARD:
        view.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS);
        break;
      case LIGHT_IMPACT:
        view.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY);
        break;
      case MEDIUM_IMPACT:
        view.performHapticFeedback(HapticFeedbackConstants.KEYBOARD_TAP);
        break;
      case HEAVY_IMPACT:
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
          view.performHapticFeedback(HapticFeedbackConstants.CONTEXT_CLICK);
        }
        break;
      case SELECTION_CLICK:
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
          view.performHapticFeedback(HapticFeedbackConstants.CLOCK_TICK);
        }
        break;
    }
  }

  @Override
  public void setPreferredOrientations(int androidOrientation) {
    setSystemChromePreferredOrientations(androidOrientation);
  }

  @Override
  public void setApplicationSwitcherDescription(
      @NonNull PlatformChannel.AppSwitcherDescription description) {
    setSystemChromeApplicationSwitcherDescription(description);
  }

  @Override
  public void showSystemOverlays(@NonNull List<PlatformChannel.SystemUiOverlay> overlays) {
    setSystemChromeEnabledSystemUIOverlays(overlays);
  }

  @Override
  public void restoreSystemUiOverlays() {
    restoreSystemChromeSystemUIOverlays();
  }

  @Override
  public void setSystemUiOverlayStyle(
      @NonNull PlatformChannel.SystemChromeStyle systemUiOverlayStyle) {
    setSystemChromeSystemUIOverlayStyle(systemUiOverlayStyle);
  }

  @Override
  public void popSystemNavigator() {
    finish();
  }

  @Nullable
  @Override
  public CharSequence getClipboardData(@Nullable PlatformChannel.ClipboardContentFormat format) {
    ClipboardManager clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
    if (!clipboard.hasPrimaryClip()) return null;

    ClipDescription clipDescription = clipboard.getPrimaryClipDescription();
 
    if (clipDescription == null || !clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN))) {
      return null;
    } 

    ClipData clip = clipboard.getPrimaryClip();

    if (format == null || format == PlatformChannel.ClipboardContentFormat.PLAIN_TEXT) {
      return clip.getItemAt(0).coerceToText(getActivity());
    }

    return null;
  }

  @Override
  public void setClipboardData(@NonNull String text) {
    ClipboardManager clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
    ClipData clip = ClipData.newPlainText("text label?", text);
    clipboard.setPrimaryClip(clip);
  }

  @Override
  public boolean clipboardHasStrings() {
    CharSequence data = getClipboardData(PlatformChannel.ClipboardContentFormat.PLAIN_TEXT);
    return data != null && data.length() > 0;
  }

  private void setSystemChromePreferredOrientations(int androidOrientation) {
    setRequestedOrientation(androidOrientation);
  }

  @SuppressWarnings("deprecation")
  private void setSystemChromeApplicationSwitcherDescription(
      PlatformChannel.AppSwitcherDescription description) {
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
      return;
    }

    // Linter refuses to believe we're only executing this code in API 28 unless we use distinct if
    // blocks and
    // hardcode the API 28 constant.
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P
        && Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {
      setTaskDescription(
          new ActivityManager.TaskDescription(
              description.label, /*icon=*/ null, description.color));
    }
    if (Build.VERSION.SDK_INT >= 28) {
      ActivityManager.TaskDescription taskDescription =
          new ActivityManager.TaskDescription(description.label, 0, description.color);
      setTaskDescription(taskDescription);
    }
  }

  private void setSystemChromeEnabledSystemUIOverlays(
      List<PlatformChannel.SystemUiOverlay> overlaysToShow) {
    // Start by assuming we want to hide all system overlays (like an immersive game).
    int enabledOverlays =
        DEFAULT_SYSTEM_UI
            | View.SYSTEM_UI_FLAG_FULLSCREEN
            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION;

    // The SYSTEM_UI_FLAG_IMMERSIVE_STICKY flag was introduced in API 19, so we apply it
    // if desired, and if the current Android version is 19 or greater.
    if (overlaysToShow.size() == 0 && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
      enabledOverlays |= View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
    }

    // Re-add any desired system overlays.
    for (int i = 0; i < overlaysToShow.size(); ++i) {
      PlatformChannel.SystemUiOverlay overlayToShow = overlaysToShow.get(i);
      switch (overlayToShow) {
        case TOP_OVERLAYS:
          enabledOverlays &= ~View.SYSTEM_UI_FLAG_FULLSCREEN;
          break;
        case BOTTOM_OVERLAYS:
          enabledOverlays &= ~View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION;
          enabledOverlays &= ~View.SYSTEM_UI_FLAG_HIDE_NAVIGATION;
          break;
      }
    }

    mEnabledOverlays = enabledOverlays;
    updateSystemUiOverlays();
  }

  /**
   * Refreshes Android's window system UI (AKA system chrome) to match Flutter's desired {@link
   * PlatformChannel.SystemChromeStyle}.
   *
   * <p>Updating the system UI Overlays is accomplished by altering the decor view of the {@link
   * Window} associated with the Activity that was provided to this {@code PlatformPlugin}.
   */
  public void updateSystemUiOverlays() {
    getWindow().getDecorView().setSystemUiVisibility(mEnabledOverlays);
    if (currentTheme != null) {
      setSystemChromeSystemUIOverlayStyle(currentTheme);
    }
  }

  private void restoreSystemChromeSystemUIOverlays() {
    updateSystemUiOverlays();
  }

  private void setSystemChromeSystemUIOverlayStyle(
      PlatformChannel.SystemChromeStyle systemChromeStyle) {
    Window window = getWindow();
    View view = window.getDecorView();
    int flags = view.getSystemUiVisibility();
    // You can change the navigation bar color (including translucent colors)
    // in Android, but you can't change the color of the navigation buttons until Android O.
    // LIGHT vs DARK effectively isn't supported until then.
    // Build.VERSION_CODES.O
    if (Build.VERSION.SDK_INT >= 26) {
      if (systemChromeStyle.systemNavigationBarIconBrightness != null) {
        switch (systemChromeStyle.systemNavigationBarIconBrightness) {
          case DARK:
            // View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
            flags |= 0x10;
            break;
          case LIGHT:
            flags &= ~0x10;
            break;
        }
      }
      if (systemChromeStyle.systemNavigationBarColor != null) {
        window.setNavigationBarColor(systemChromeStyle.systemNavigationBarColor);
      }
    }
    // Build.VERSION_CODES.M
    if (Build.VERSION.SDK_INT >= 23) {
      if (systemChromeStyle.statusBarIconBrightness != null) {
        switch (systemChromeStyle.statusBarIconBrightness) {
          case DARK:
            // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
            flags |= 0x2000;
            break;
          case LIGHT:
            flags &= ~0x2000;
            break;
        }
      }
      if (systemChromeStyle.statusBarColor != null) {
        window.setStatusBarColor(systemChromeStyle.statusBarColor);
      }
    }
    if (systemChromeStyle.systemNavigationBarDividerColor != null) {
      // Not available until Android P.
      // window.setNavigationBarDividerColor(systemNavigationBarDividerColor);
    }
    view.setSystemUiVisibility(flags);
    currentTheme = systemChromeStyle;
  }
}
```

**Comment by b055man** [comment]
Thanks! 
Can you link the pull request with the change/commit so that it is possible track when it eventually is available in the stable release?

**Comment by hamdikahloun** [comment]
I will do it soon.

**Comment by hamdikahloun** [comment]
## Steps to Reproduce



1. Run `flutter create -a java clipboard`.
2. Update MainActivity.java as follows: 
<details>
  <summary>MainActivity.java</summary>

```
import android.app.AlertDialog;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import androidx.annotation.Nullable;
import io.flutter.embedding.android.FlutterActivity;
import static android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI;

public class MainActivity extends FlutterActivity {
    private static int LOAD_IMAGE_RESULTS = 1;
    private ClipboardManager clipboard;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Intent i = new Intent(Intent.ACTION_PICK, EXTERNAL_CONTENT_URI);
        clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
        startActivityForResult(i, LOAD_IMAGE_RESULTS);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == LOAD_IMAGE_RESULTS && resultCode == RESULT_OK && data != null) {

            Uri pickedImage = data.getData();

            ClipData clip = ClipData.newUri(getContentResolver(), "URI", pickedImage);
            clipboard.setPrimaryClip(clip);

            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
            builder.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
                public void onClick(DialogInterface dialog, int id) {
                    dialog.dismiss();
                }
            });

            builder.setMessage("You can close this application now.")
                    .setTitle("Flutter");
            AlertDialog dialog = builder.create();
            dialog.show();
        }
    }
}

```
</details>

3. `flutter create bug`

4. Update main.dart as follows: 

<details>
  <summary>main.dart</summary>

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Retrieve Text Input',
      home: MyCustomForm(),
    );
  }
}

class MyCustomForm extends StatefulWidget {
  @override
  _MyCustomFormState createState() => _MyCustomFormState();
}

class _MyCustomFormState extends State<MyCustomForm> {
  final myController = TextEditingController();

  @override
  void initState() {
    super.initState();

    myController.addListener(_printLatestValue);
  }

  @override
  void dispose() {
    myController.dispose();
    super.dispose();
  }

  _printLatestValue() {
    print("Second text field: ${myController.text}");
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Retrieve Text Input'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: <Widget>[
            TextField(
              controller: myController,
            ),
          ],
        ),
      ),
    );
  }
}
```
</details>

5. Start clipboard application :  `flutter run`

6. Start bug application : `flutter run`

<details>
  <summary>Logs</summary>



```
flutter run
Using hardware rendering with device Android SDK built for x86. If you notice graphics artifacts, consider enabling
software rendering with "--enable-software-rendering".
Launching lib\main.dart on Android SDK built for x86 in debug mode...
Running Gradle task 'assembleDebug'...
Running Gradle task 'assembleDebug'... Done                         8,8s
√ Built build\app\outputs\flutter-apk\app-debug.apk.
Waiting for Android SDK built for x86 to report its views...         17ms
I/com.example.bu( 6937): Background young concurrent copying GC freed 26057(6576KB) AllocSpace objects, 0(0B) LOS objects, 84% free, 1136KB/7280KB, paused 1.969ms total 123.789ms
D/EGL_emulation( 6937): eglMakeCurrent: 0xe0132120: ver 2 0 (tinfo 0xe360f730)
Syncing files to device Android SDK built for x86...               345ms

Flutter run key commands.
r Hot reload.
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:57480/Mlx6TRtoAOA=/
E/MethodChannel#flutter/platform( 6937): Failed to handle method call
E/MethodChannel#flutter/platform( 6937): java.lang.SecurityException: UID 10133 does not have permission to content://com.google.android.apps.photos.contentprovider/-1/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F26/ORIGINAL/NONE/748929461 [user 0]
E/MethodChannel#flutter/platform( 6937):        at android.os.Parcel.createException(Parcel.java:2071)
E/MethodChannel#flutter/platform( 6937):        at android.os.Parcel.readException(Parcel.java:2039)
E/MethodChannel#flutter/platform( 6937):        at android.os.Parcel.readException(Parcel.java:1987)
E/MethodChannel#flutter/platform( 6937):        at android.content.IClipboard$Stub$Proxy.getPrimaryClip(IClipboard.java:333)
E/MethodChannel#flutter/platform( 6937):        at android.content.ClipboardManager.getPrimaryClip(ClipboardManager.java:135)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.plugin.platform.PlatformPlugin.getClipboardData(PlatformPlugin.java:274)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.plugin.platform.PlatformPlugin.access$700(PlatformPlugin.java:24)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.plugin.platform.PlatformPlugin$1.getClipboardData(PlatformPlugin.java:81)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.embedding.engine.systemchannels.PlatformChannel$1.onMethodCall(PlatformChannel.java:141)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:230)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/MethodChannel#flutter/platform( 6937):        at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
E/MethodChannel#flutter/platform( 6937):        at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#flutter/platform( 6937):        at android.os.MessageQueue.next(MessageQueue.java:336)
E/MethodChannel#flutter/platform( 6937):        at android.os.Looper.loop(Looper.java:174)
E/MethodChannel#flutter/platform( 6937):        at android.app.ActivityThread.main(ActivityThread.java:7356)
E/MethodChannel#flutter/platform( 6937):        at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#flutter/platform( 6937):        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/MethodChannel#flutter/platform( 6937):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
E/MethodChannel#flutter/platform( 6937): Caused by: android.os.RemoteException: Remote stack trace:
E/MethodChannel#flutter/platform( 6937):        at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermission(UriGrantsManagerService.java:1179)
E/MethodChannel#flutter/platform( 6937):        at com.android.server.uri.UriGrantsManagerService.grantUriPermission(UriGrantsManagerService.java:767)
E/MethodChannel#flutter/platform( 6937):        at com.android.server.uri.UriGrantsManagerService.grantUriPermissionFromOwner(UriGrantsManagerService.java:208)
E/MethodChannel#flutter/platform( 6937):        at com.android.server.clipboard.ClipboardService.grantUriLocked(ClipboardService.java:655)
E/MethodChannel#flutter/platform( 6937):        at com.android.server.clipboard.ClipboardService.grantItemLocked(ClipboardService.java:670)
```



```
flutter analyze
Analyzing bug...
No issues found! (ran in 20.9s)
```



```
flutter doctor -v
[√] Flutter (Channel stable, 1.20.4, on Microsoft Windows [version 10.0.19041.508], locale fr-FR)
    • Flutter version 1.20.4 at D:\flutter_windows_1.20.4-stable\flutter
    • Framework revision fba99f6cf9 (13 days ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2


[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at C:\Users\hamdi\AppData\Local\Android\Sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = C:\Users\hamdi\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.49.2)
    • VS Code at C:\Users\hamdi\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.14.1

[√] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

• No issues found!
```

</details>
